a6/app/views.py

```diff
@@ -0,0 +1,251 @@
+
+    def create(self, request, *args, **kwargs):
+        orderSerializer = OrderSerializer(data=request.data)
+        data = request.data
+
+        if orderSerializer.is_valid():
+            username = validated_data['uid']['company_name'] if 'company_name' in validated_data['uid'] else None
+            address = validated_data['uid']['addr_line1'] if 'addr_line1' in validated_data['uid'] else None
+            status_str = data['status']
+
+            # if username is not None:
+            #       Create User if not exists
+            # else get user
+            if username:
+                try:
+                    # maybe, a better functionality would be to get user based on name as well as address
+                    user = User.objects.get(company_name=username)
+                except:
+                    user = User(company_name=username, addr_line1=address)
+                    user.save()
+            else:
+                user = User.objects.get_or_create(company_name="")[0]
+
+            if status_str.lower() == "created":
+                try:
+                    order = Orders(uid=user, status=status_str, timestamp=datetime.datetime.now())
+                    order.save()
+                except:
+                    # there was some problem with save
+                    return Response({}, status=status.HTTP_500_INTERNAL_SERVER_ERRORs)
+            else:
+                order = Orders(uid=user, status=status_str)
+                order.save()
+
+            serializer = OrderSerializer(order)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        else:
+            return Response(None, status=status.HTTP_400_BAD_REQUEST)
+
+
+
+class OrderLineViewSet(mixins.RetrieveModelMixin, mixins.CreateModelMixin, mixins.ListModelMixin, viewsets.GenericViewSet):
+    queryset = OrderProduct2.objects.filter(order__deleted=0)
+    serializer_class = OrderLineItemSerializer
+
+    def create(self, request, *args, **kwargs):
+        orderLineItemSerializer = OrderLineItemSerializer(data=request.data)
+        data = request.data
+        if orderLineItemSerializer.is_valid():
+            product = Product.objects.get(id=int(data['product_id']))
+            order = Orders.objects.get(oid=int(kwargs['order_id']))
+            try:
+                order_product = OrderProduct2.objects.get(product=product, order=order)
+                order_product.selling_cost = round(float(data['price']), 4)
+                order_product.save()
+            except:
+                order_product = OrderProduct2(product=product, order=order, buying_cost=round(float(data['price']), 4))
+                order_product.save()
+            serializer = OrderLineItemSerializer(order_product)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            return Response(None, status=status.HTTP_400_BAD_REQUEST)
+
+    def retrieve(self, request, *args, **kwargs):
```

Comment:
Viewset automatically takes care of the retrieve and list commands and we need not overwrite it for simple fetch. Since the queryset has taken care of the deleted orders, this function was not necessary in my opinion.